python/cudf/cudf/io/parquet.py

```diff
@@ -680,7 +702,29 @@ def to_parquet(
     """{docstring}"""
 
     if engine == "cudf":
+        # Ensure that no columns dtype is 'category'
+        for col in df.columns:
+            if df[col].dtype.name == "category":
+                raise ValueError(
+                    "'category' column dtypes are currently not "
+                    + "supported by the gpu accelerated parquet writer"
+                )
```

Comment:
Not necessary to address anything in this Pr, but it seems like we should be able to handle categorical dtypes for the partitioned columns.